### PR TITLE
Adds A Unique Objective For Arcfiends

### PR DESCRIPTION
**Pull request by Wisemonster**
[FEATURE]

## About the PR 
This pr adds an objective for arcfiends to go for, involving draining a total of a certain amount of power.

## Why's this needed? 
Currently, arcfiends do not have any unique objectives. This fixes that by giving them objectives that fit them.

## Changelog

```changelog
(u)Wisemonster
(*)Add a new objective for Arcfiends!
```

### CODE REVIEW

**Comment by Wisemonster** [comment]
Wasn't sure how high I should set the arcfiend objective goals. Let me know if it's too high/low.

**Comment by Tarmunora** [comment]
@Sovexe

File: code/modules/antagonists/arcfiend/arcfiend.dm [hunk 1]
```diff
@@ -47,6 +56,12 @@
 		var/points = min((MAX_ARCFIEND_POINTS - src.points), add_points)
 		. = ..(points, target_ah_type)
 
+	proc/get_arcfiend_power(var/total_power = 0)
+		if (total_power)
+			return src.lifetime_energy
+		else
+			return src.points
+
```
Comment (Sovexe) [inline code/modules/antagonists/arcfiend/arcfiend.dm:64]: should be two procs, but at that point it also has no point vs just checking vars
```suggestion
```

File: code/modules/antagonists/arcfiend/arcfiend.dm [hunk 2]
```diff
@@ -28,6 +28,15 @@
...
 
+/mob/proc/get_arcfiend_power(var/lifetime_energy = 0)
+	if (!isarcfiend(src))
+		return 0
+
+	var/datum/abilityHolder/arcfiend/AH = src.get_ability_holder(/datum/abilityHolder/arcfiend)
+	if (AH && istype(AH))
+		return AH.get_arcfiend_power(lifetime_energy)
+	else
+		return 0
```
Comment (Sovexe) [inline code/modules/antagonists/arcfiend/arcfiend.dm:39]: don't want to add more procs to mob
```suggestion
```

File: code/datums/objective.dm [hunk 3]
```diff
@@ -1389,6 +1389,27 @@ ABSTRACT_TYPE(/datum/objective/conspiracy)
...
+#else
+		powercount = rand(210,250) * 10
+#endif
+		explanation_text = "Accumulate at least [powercount] units of charge in total."
+
+	check_completion()
+		if (owner.current && owner.current.get_arcfiend_power(1) >= powercount)
+			return 1
+		else
+			return 0
```
Comment (Sovexe) [inline code/datums/objective.dm:1411]:
```suggestion
	check_completion()
		var/datum/abilityHolder/arcfiend/AH = owner.current?.get_ability_holder(/datum/abilityHolder/arcfiend)
		return (AH?.lifetime_energy >= powercount)
```

File: code/datums/objective.dm [hunk 4]
```diff
@@ -1389,6 +1389,27 @@ ABSTRACT_TYPE(/datum/objective/conspiracy)
 
 		return 1
 
+/////////////////////////////////////////////////////////
+// Arcfiend Objectives                                 //
+/////////////////////////////////////////////////////////
+
+/datum/objective/specialist/powerdrain // this is basically just a repurposed vamp objective, but it should work.
+	var/powercount
```
Comment (Sovexe) [inline code/datums/objective.dm:1397]: should probably called like power goal or something, since its a target not your count

**Review by Sovexe — CHANGES_REQUESTED** [review]

**Comment by Sovexe** [comment]
Power goals need to be higher, while 2500 is the cap it's not all that hard to reach 

Also could use additional goal variety if you're up for it. One idea would be to have a certain station area drained of power

**Comment by Wisemonster** [comment]
I'll look into adding in an area power denial objective, although I'm not sure how to make it atm so that it requires more effort than simply removing an areas cell from the apc.

**Comment by Wisemonster** [comment]
I might try to add the apc goal in a separate pr, cause I'm having trouble getting it to work.

**Pull request merged**